benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/scheduler/TestTaskScheduler.java

```diff
@@ -233,12 +240,48 @@ private void addStoredKnowledge(String testID) {
     AddStoredKnowledgeTask task = new AddStoredKnowledgeTask(testID);
 
     // TODO - should go into a stateless queue (so that we can recover)
-    Future<?> future = taskExecutorService.submit(task);
+    Future<Boolean> future = taskExecutorService.submit(task);
```

Comment:
Done in https://github.com/benchflow/benchflow/pull/394/commits/ab816423446799a9f6e388d52f85c15bef9a4c58